pkg/domain/infra/abi/play.go

```diff
@@ -267,12 +267,9 @@ func (ic *ContainerEngine) playKubePod(ctx context.Context, podName string, podY
 	}
 
 	if podOpt.Infra {
-		imagePull := config.DefaultInfraImage
-		if podOpt.InfraImage != config.DefaultInfraImage && podOpt.InfraImage != "" {
```

Comment:
Just a general comment @chenzhiwei , this seems like something you should be changing in `kube.ToPodOpt` not within this kube generation code. The issue seems to be that if specified in the YAML, the InfraImage is not being specified properly. That is a `ToPodOpt` issue

Reply:
The reason I change there is because the logic is here.